src/engine/strat_engine/mdv.rs

```diff
@@ -33,6 +33,32 @@ const DEV_PATH: &'static str = "/dev/stratis";
 
 const FILESYSTEM_DIR: &'static str = "filesystems";
 
+fn with_mount<F, O>(dev: &Path, mount_point: &Path, op: F) -> EngineResult<O>
+    where F: FnOnce() -> EngineResult<O>
+{
+    if let Err(err) = mount(Some(dev),
+                            mount_point,
+                            Some("xfs"),
+                            MsFlags::empty(),
+                            None as Option<&str>) {
+        match err {
+            nix::Error::Sys(nix::Errno::EBUSY) => {
+                // EBUSY is the error returned in this context when the
+                // device is already mounted at the specified mountpoint.
+            }
+            _ => return Err(From::from(err)),
+        }
+    };
+
+    let res = op().map_err(|err| err.into());
+
+    umount(mount_point).unwrap_or_else(|err| {
+                                           error!("Could not unmount {:?}: {}", mount_point, err)
```

Comment:
Failure to unmount may be an error for unmount, but it is definitely not an error for stratisd.

Reply:
`warn!()`?